plugin/web/extensions/location/modals/commonModalWrapper/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { Divider, Button, Icon } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { Children, Fragment, ReactNode } from "react";
+
+export type Props = {
+  title?: string;
+  children?: ReactNode;
+  onModalChange: () => void;
+};
+
+const CommonModalWrapper: React.FC<Props> = ({ title, children, onModalChange }) => {
+  const childArray = Children.toArray(children);
+
+  return (
+    <Wrapper>
+      {title && (
+        <>
+          <HeaderWrapper>
+            <Title>{title}</Title>
+            <CloseButton>
+              <Icon icon="close" onClick={onModalChange} size={16} />
+            </CloseButton>
+          </HeaderWrapper>
+          <Divider />
+        </>
+      )}
+      {childArray.map((child, idx) => (
+        <FragmentWrapper key={idx}>
+          {child}
+          {idx + 1 !== childArray.length && <span />}
+        </FragmentWrapper>
+      ))}
+
+      <FooterWrapper>
+        <OkButton>
+          <Icon icon="close" onClick={onModalChange} />
+        </OkButton>
+      </FooterWrapper>
+    </Wrapper>
+  );
+};
+
+export default CommonModalWrapper;
+
+const Wrapper = styled.div`
+  padding: 32px 16px;
+`;
+
+const Title = styled.p`
+  font-size: 16px;
+`;
+const CloseButton = styled(Button)`
+  margin-right: 0px;
+  width: 16px;
+  height: 16px;
+`;
+const HeaderWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding: 16px 24px;
+  gap: 36px;
+  width: 572px;
+  height: 56px;
+  background: #ffffff;
+  box-shadow: inset 0px -1px 0px #f0f0f0;
+  flex: none;
+  order: 0;
+  align-self: stretch;
+  flex-grow: 0;
+`;
+const FragmentWrapper = styled(Fragment)`
```

Comment:
I am trying to make general as much as I can and also because I have some many type of child component inside the modals

Reply:
I removed this fragment now but my idea was to keep it general

Reply:
386626f0ec753cdcae932611366d0e17a73470a4

Reply:
Fragments are useful to avoid many unnecessary nested divs. In this case, we need to add styles, so it is a legitimate wrapper. If you are just using a div to group elements and that is it, and you already have quite a bit of nesting, then fragments are useful.

Reply:
You can resolve this comment after you read!